src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String[] STATISTICS_FIELDS = {"supply", "buy", "result"};
+    private static final int SUPPLY_POS = 0;
+    private static final int BUY_POS = 1;
+    private static final int RESULT_POS = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(readFromFile(fromFileName), toFileName);
+    }
+
+    private int[] readFromFile(String fromFileName) {
+        int[] statisticsValues = new int[3];
+
+        try (FileReader reader = new FileReader(fromFileName);
+                BufferedReader bufferedReader = new BufferedReader(reader)) {
+            String readLine = bufferedReader.readLine();
+            String[] readValue;
+
+            while (readLine != null) {
+                readValue = readLine.split(",");
+                if (readValue[0].equals(STATISTICS_FIELDS[SUPPLY_POS])) {
+                    statisticsValues[0] += Integer.parseInt(readValue[1]);
+                } else {
+                    statisticsValues[1] += Integer.parseInt(readValue[1]);
+                }
+                readLine = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error while reading a file");
```

It is better to throw the exception further, and so we displayed it on the screen, and then some logic is done and the result is returned, but it should not

---

not fixed